rust/agama-server/src/web/common/issues.rs

```diff
@@ -0,0 +1,280 @@
+    SendCommand,
+    #[error("Error parsing issues from D-Bus: {0}")]
+    InvalidIssue(#[from] zbus::zvariant::Error),
+    #[error("Error reading the issues: {0}")]
+    DBus(#[from] zbus::Error),
+    #[error("Invalid D-Bus name: {0}")]
+    DBusName(#[from] zbus::names::Error),
+    #[error("Could not send the event: {0}")]
+    SendEvent(#[from] broadcast::error::SendError<Event>),
+}
+
+#[derive(Debug)]
+enum IssuesCommand {
+    Get(String, String, oneshot::Sender<Vec<Issue>>),
+}
+
+/// Implements a Tokio task that holds the issues for each service.
+pub struct IssuesService {
+    issues: HashMap<String, Vec<Issue>>,
+    commands: mpsc::Receiver<IssuesCommand>,
+    events: EventsSender,
+    dbus: zbus::Connection,
+}
+
+impl IssuesService {
+    /// Sets up and starts the service as a Tokio task.
+    ///
+    /// Once it is started, the service waits for:
+    ///
+    /// * Commands from a client ([IssuesClient]).
+    /// * Relevant events from D-Bus.
+    pub async fn start(dbus: zbus::Connection, events: EventsSender) -> IssuesClient {
+        let (tx, rx) = mpsc::channel(16);
+        let mut service = IssuesService {
+            issues: HashMap::new(),
+            dbus,
+            events,
+            commands: rx,
+        };
+
+        tokio::spawn(async move { service.run().await });
+        IssuesClient(tx)
+    }
+
+    /// Main loop of the service.
+    async fn run(&mut self) {
+        let mut messages = build_properties_changed_stream(&self.dbus).await.unwrap();
+        loop {
+            tokio::select! {
+                Some(cmd) = self.commands.recv() => {
+                    if let Err(e) = self.handle_command(cmd).await {
+                        tracing::error!("{e}");
+                    }
+                }
+
+                Some(Ok(message)) = messages.next() => {
+                    if let Some(changed) = PropertiesChanged::from_message(message) {
+                        if let Err(e) = self.handle_property_changed(changed) {
+                            tracing::error!("IssuesService: could not handle change: {:?}", e);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /// Handles commands from the client.
+    async fn handle_command(&mut self, command: IssuesCommand) -> IssuesServiceResult<()> {
+        match command {
+            IssuesCommand::Get(service, path, tx) => {
+                let issues = self.get(&service, &path).await?;
+                tx.send(issues)
+                    .map_err(|_| IssuesServiceError::SendIssues)?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Handles PropertiesChanged events.
+    fn handle_property_changed(&mut self, message: PropertiesChanged) -> IssuesServiceResult<()> {
+        let args = message.args()?;
+        let inner = message.message();
+        let header = inner.header();
+        let Some(path) = header.path() else {
+            return Ok(());
```

Comment:
I think I need comment here why it is Ok to not have header path.

Reply:
It is not an error, it is simply what we are not interested in this message.